collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.kt

```diff
@@ -97,6 +99,18 @@ class ApplicationInitializer(
     }
 
     private fun initializeMapFrameworks() {
+        // Reset basemap setting if the currently selected is not available
```

Comment:
If we install the app this code will be executed but what if we scan a QR code with maps that are not available?

Reply:
Ah good point. The QR code map would be set and used (even if it's not available), but it would be then be reset the next time the app is launched (as a fresh process). This could cause a crash I'd imagine if you used a QR code with Mapbox on a build without Mapbox and would give you the blank screen case for Google Maps (if there's no Google Maps key). I'm not sure whether this is worth fixing seeing as we expect all maps to be available in full release builds. What do you think?

Reply:
Can't we just run that code or a part of it after loading new settings to update maps?

Reply:
Yeah I guess we can pull this code to something shared and call it on settings change as well.

Reply:
So we can have a separate issue for that improvement? We can ask the QA team to verify this case to make sure it works in the way we assume and then file an issue.

Reply:
Yeah it's actually a little awkward to get that working right now as we'd need to rework things so that `CollectSettingsChangeHandler` is able to actually change settings. I reckon we leave it for now and come back for it it gets annoying.